src/main/java/org/jenkinsci/plugins/gitclient/CliGitAPIImpl.java

```diff
@@ -1681,44 +1695,39 @@ private File getSSHExeFromGitExeParentDir(String userGitExe) {
         throw new RuntimeException("ssh executable not found. The git plugin only supports official git client http://git-scm.com/download/win");
     }
 
-    private File createWindowsGitSSH(File key, String user) throws IOException {
-        File ssh = File.createTempFile("ssh", ".bat");
+    private FilePath createWindowsGitSSH(FilePath key, String user) throws IOException, InterruptedException {
+        FilePath ssh = workspace.createTempFile("ssh", ".bat");
```

Comment:
not insecure - but no need to clutter the workspace.
use [`WorkspaceList.tmpDir(workspace)`](https://github.com/jenkinsci/jenkins/commit/307bfc17385c14aefcb623ccc3e7144e4f01e2a3)

If you can not bump a version to that write a helper that does the same thing.